src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -327,15 +327,15 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
                 + " Run id '{}'", hostName, segment.getRunId());
       JmxProxy hostProxy = null;
       try{
-        Optional<HostMetrics> hostMetrics;
+        Optional<HostMetrics> hostMetrics = Optional.absent();
         try{
           hostProxy = context.jmxConnectionFactory.connect(hostName);
           connected = true;
+          hostMetrics = getMetricsForHost(hostName, hostProxy);
```

Comment:
Moving this here won't work because if we cannot connect to one host through JMX (which will happen in multi region clusters) then Reaper won't look if metrics are already there in storage, which it does here : https://github.com/thelastpickle/cassandra-reaper/blob/ft-reaper-improvements-final/src/main/java/com/spotify/reaper/service/SegmentRunner.java#L417-L436

The error are only displayed in DEBUG so it shouldn't clutter the logs. 
In which conditions did you get a NPE ?